src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -690,6 +718,24 @@ mp::MountHandler::UPtr mp::QemuVirtualMachine::make_native_mount_handler(const s
     return std::make_unique<QemuMountHandler>(this, &key_provider, target, mount);
 }
 
+void mp::QemuVirtualMachine::remove_snapshots_from_image() const
```

Comment:
I think this function should be part of `qemu_img_utils`

Reply:
`QemuVirtualMachine::remove_snapshots_from_image` is a derived function from the interface. Do you mean we should move this function into `qemu_img_utils`? Or you meant some lines of the code below can be wrapped up into qemu_img_utils backend namespace? For example, like `bool instance_image_has_snapshot(const Path& image_path, QString snapshot_tag);`.

Reply:
Hmmm, `instance_image_has_snapshot` is used by several hypervisors so it would make sense to keep the function. It's more so the contents of this function that I am seeing a lot of similarities/overlap with other functions in `qemu_img_utils`.

Reply:
I moved `snapshot -l` and `snapshot -d` code into functions, indeed, it is a small improvement.